src/tests/ftest/user_setup.py

```diff
@@ -0,0 +1,137 @@
+
+    '''
+    return ('sudo ' if sudo else '') + command
+
+
+def create_group(nodes, group, sudo=False):
+    '''Create a group on remote nodes.
+
+    Args:
+        nodes (NodeSet): nodes on which to create the group
+        group (str): the group to create
+        sudo (bool): whether to execute commands with sudo
+
+    Returns:
+        bool: whether the command was successful on all nodes
+    '''
+    command = _sudo(sudo, 'groupadd {} -f'.format(group))
+    return run_remote(LOGGER, nodes, command).passed
+
+
+def create_users(nodes, users, group=None, parent_dir=None, sudo=False):
+    '''Create some users.
+
+    Args:
+        nodes (NodeSet): nodes on which to create the users
+        users (list): users to create
+        group (str, optional): user group. Default is None
+        parent_dir (str, optional): parent home directory
+        sudo (bool): whether to execute commands with sudo
+
+    Returns:
+        bool: whether the command was successful on all nodes
+    '''
+    for user in users:
+        # Delete if existing
+        check_cmd = 'id -u {} &> /dev/null'.format(user)
+        delete_cmd = _sudo(sudo, 'userdel -f -r {}'.format(user))
+        command = 'if $({}); then {}; fi'.format(check_cmd, delete_cmd)
+        if not run_remote(LOGGER, nodes, command).passed:
+            return False
+
+        # Create new user
+        command = ' '.join(filter(None, [
+            _sudo(sudo, 'useradd -m'),
+            '-g {}'.format(group) if group else None,
+            '-d {}'.format(os.path.join(parent_dir, user)) if parent_dir else None,
+            user
+        ]))
+        if not run_remote(LOGGER, nodes, command).passed:
+            return False
+    return True
+
+
+def main():
+    '''Set up test env users.
+
+    Returns:
+        int: 0 on success. positive int on failure
+    '''
+    logging.basicConfig(
+        format="%(asctime)s %(levelname)-5s %(message)s",
+        datefmt=r"%Y/%m/%d %I:%M:%S", level=logging.DEBUG)
+
+    parser = argparse.ArgumentParser(prog="user_setup.py")
+
+    parser.add_argument(
+        '-n', '--nodes',
+        type=str,
+        help="Comma-separated list of nodes to create users on")
+    parser.add_argument(
+        '-u', '--users',
+        type=str,
+        required=True,
+        help="Comma-separated list of user accounts to create")
+    parser.add_argument(
+        "-g", "--group",
+        type=str,
+        help="Group to add users to. Will be created if non-existent")
+    parser.add_argument(
+        "-d", "--dir",
+        type=str,
+        help="Parent home directory for users")
+    parser.add_argument(
+        "-s", "--sudo",
+        action="store_true",
+        help="Run all commands with privileges")
+
+    args = parser.parse_args()
+    logging.info("Arguments: %s", args)
+
+    nodes = NodeSet(args.nodes or get_local_host())
+    users = args.users.split(',') if args.users else None
```

Comment:
This should either default to empty list (so you don't have a problem on line 67) or the progarm should quit if no users are specified:
```suggestion
    users = args.users.split(',') if args.users else []
```

Alternatively, the string `split()` could be avoided by defining the `--user` argument with `nargs="*"` and then `required=True` should ensure at least one user was provided.

Reply:
nargs="*" means a list of zero or more, nargs="+" means a list of one or more which I think is what you're looking for here.

Reply:
Since the name of the script is `user_setup.py`, I changed this to a positional and used `nargs='+',`